decidim-surveys/db/migrate/20240925124312_add_settings_to_decidim_surveys_surveys.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class AddSettingsToDecidimSurveysSurveys < ActiveRecord::Migration[7.0]
+  class Survey < ApplicationRecord
+    self.table_name = :decidim_surveys_surveys
+  end
+
+  def change
+    reversible do |dir|
+      dir.up do
+        add_column :decidim_surveys_surveys, :starts_at, :datetime
+        add_column :decidim_surveys_surveys, :ends_at, :datetime
+        add_column :decidim_surveys_surveys, :announcement, :jsonb
+        add_column :decidim_surveys_surveys, :allow_answers, :boolean
+        add_column :decidim_surveys_surveys, :allow_unregistered, :boolean
+        add_column :decidim_surveys_surveys, :clean_after_publish, :boolean
+        add_column :decidim_surveys_surveys, :published_at, :datetime
+        add_index :decidim_surveys_surveys, :published_at
+
+        Survey.where(published_at: nil).find_each do |survey|
+          published_at = survey.component.published_at || survey.created_at
```

Comment:
If the component's published_at is nil, it means that respective survey has not been published. Mind that applying this, may raise a rubocop error. 

```suggestion
          published_at = survey.component.published_at
          next if published_at.nil?
```

Reply:
Done